manager/filters.py

```diff
@@ -599,6 +600,25 @@ def _filter_inventory_group_ids(query, args, _kwargs):
     return query
 
 
+def _filter_system_by_host_type(query, args, _kwargs):
+    """
+    Filters systems by their host type.
+    Query needs to be joined with system_platform table!
+
+    Args:
+        query (object): Query object to apply filters to
+        args (dict): Query arguemnts
+
+    Returns:
+        object: Modified query with host type filter applied
+    """
+    if host_types := args.get("host_type", [HostType.RPMDNF]):
+        if HostType.RPMDNF in host_types and HostType.EDGE in host_types:
```

Comment:
what about something like:
```suggestion
        if all(enum in host_types for enum in list(HostType)):
```